backend/TRFSAE.MemberPortal.API/Controllers/PurchaseRequestController.cs

```diff
@@ -0,0 +1,54 @@
+using Microsoft.AspNetCore.Mvc;
+using TRFSAE.MemberPortal.API.DTOs;
+using TRFSAE.MemberPortal.API.Interfaces;
+using Supabase;
+
+namespace TRFSAE.MemberPortal.API.Controllers
+{
+    [ApiController]
+    [Route("api/purchase-request")]
+    public class PurchaseRequestController : ControllerBase
+    {
+        private readonly IPurchaseRequestService _purchaseRequestService;
+
+        public PurchaseRequestController(IPurchaseRequestService purchaseRequestService)
+        {
+            _purchaseRequestService = purchaseRequestService;
+        }
+
+        [HttpGet]
+        public async Task<IActionResult> GetAllPurchaseAsync([FromQuery] PurchaseRequestSearchDto? dto)
+        {
+            var result = await _purchaseRequestService.GetAllPurchaseRequestsAsync(dto ?? new PurchaseRequestSearchDto());
```

Comment:
Why are we making a new DTO here? There should only exist 1 instance of a DTO as it's a singleton.